Add Debug, JSON Format options for Java Tracer logs (#30141)
DataDog/documentation: content/en/tracing/troubleshooting/tracer_debug_logs.md

To enable debug mode for the Datadog Java Tracer, set the flag `-Ddd.trace.debug=true` when starting the JVM or add `DD_TRACE_DEBUG=true` as environment variable.

**Notes**: 

> - Datadog Java Tracer implements SLF4J SimpleLogger, so [all of its settings can be applied][1], for example, logging to a dedicated log file:

Reviewer: ```suggestion
- Datadog Java Tracer implements SLF4J SimpleLogger, so [all of its settings can be applied][1]. For example, you can configure it to log to a dedicated log file:
```

```
-Ddatadog.slf4j.simpleLogger.logFile=<NEW_LOG_FILE_PATH>
```

> - To have Datadog Java Tracer logs written in a JSON format compatible out of the box with the Datadog Logs UI, use:

Reviewer: ```suggestion
- To output Datadog Java Tracer logs in a JSON format compatible with the Datadog Logs UI, use:
```
